### PR TITLE
fix the issue where files with zero length were not saved locally after a GET

### PR DESCRIPTION
**Pull request by shabtaisharon**
NETSDK-138

### CODE REVIEW

File: Ds3/Models/Range.cs [hunk 1]
```diff
@@ -21,11 +21,19 @@ public struct Range : IComparable, IComparable<Range>
...
+            if (length == 0)
+            {
+                return new Range(start, start, length);
+            }
             return new Range(start, start + length - 1, length);
         }
 
         public static Range ByPosition(long start, long end)
         {
+            if (start == 0 & end == 0)
```
Comment (RachelTucker) [inline Ds3/Models/Range.cs]: should this be &&?

Comment (shabtaisharon) [inline Ds3/Models/Range.cs]: :+1:

File: Ds3/Lang/StreamsUtil.cs [hunk 2]
```diff
@@ -25,13 +25,23 @@ public static void BufferedCopyTo(Stream src, Stream dst, int bufferSize)
             var totalBytesRead = 0;
             int bytesRead;
 
-            while ((bytesRead = src.Read(buffer, totalBytesRead, bufferSize - totalBytesRead)) != 0)
+            bytesRead = src.Read(buffer, totalBytesRead, bufferSize - totalBytesRead);
+
+            //in case the file we are getting is zero length, we still want to write it and create the file localy
+            if (bytesRead == 0)
+            {
+                dst.Write(buffer, 0, totalBytesRead);
```
Comment (rpmoore) [inline Ds3/Lang/StreamsUtil.cs]: I don't understand.  Why do we have to do the write?

Comment (shabtaisharon) [inline Ds3/Lang/StreamsUtil.cs]: This is the only way to make the stream store to open the file and create it

**Comment by shabtaisharon** [comment]
we will not fix that for 4.0.x

**Pull request closed without merging**